src/renderer/components/dock/create-resource.tsx

```diff
@@ -21,6 +26,55 @@ interface Props {
 @observer
 export class CreateResource extends React.Component<Props> {
   @observable error = "";
+  @observable templates:GroupSelectOption[] = [];
+
+  componentDidMount(){
```

Comment:
I believe all paths/templates manipulations should be done outside of this React component. Basically, all templates should be derived from store. `src/renderer/components/dock/create-resource.store.ts` is a good place to move all logic here.

Reply:
Also, please spread this code to smaller reusable methods. For example:
* parsing lens and user templates may be separated.
* there is code repetition with
```
fs.readdirSync(this.lensStore).forEach(group => {
```
```
fs.readdirSync(this.userStore).forEach((group) => {
```
This can be extracted into own method.
* we can make code smaller by keeping `React-Select` options preparation and `renderWidthBadge` inside component.